lib/routes/authorization.js

```diff
@@ -213,6 +213,23 @@ module.exports = {
           exitEarly = true;
           throw AppError.invalidAssertion();
         }
+        // Any request for a key-bearing scope should be using a verified token.
+        // Double-check that here as a defense-in-depth measure.
+        if (! claims['fxa-tokenVerified']) {
+          return P.each(scope.values(), scope => {
+            // We know only URL-format scopes can have keys,
+            // so avoid trips to the DB for common scopes like 'profile'.
+            if (scope.startsWith('https://')) {
+              return db.getScope(scope).then(s => {
+                if (s.hasScopedKeys) {
+                  throw AppError.invalidAssertion();
```

Comment:
Do you need an `exitEarly = true` [like up here](https://github.com/mozilla/fxa-oauth-server/pull/561/files#diff-422a5c70fd5a58a549ee0f442ed2e452R213)?

Reply:
Hmm...thinking about it, I wonder if that earlier `exitEarly` ever actually triggers.  Verifying assertions is pretty slow because of public-key crypto, and hitting the database is pretty fast, it wouldn't surprise me if the second are almost always wins this race.  I'm happy to add it but I'm not confident it'd have any effect in practice.

Reply:
I went ahead and added this, and also added a *check* in this first promise chain, in case the second promise chain is the one that finishes first.